site/pages/api/block-rate/index.ts

```diff
@@ -0,0 +1,28 @@
+import { NextApiRequest, NextApiResponse } from "next/types";
+import { getBlockRate } from "lib/getBlockRate";
+
+type ResponseData = {
+  blockRate30Day?: string;
+  error?: string;
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseData>
+) {
+  try {
+    // share-cache after last cached request:
+    // 0-12 hours -> fresh, return from cache
+    // 12-24 hours -> stale, return from cache but revalidate in background
+    // >24 hours -> stale, recalculate & return fresh value
+    res.setHeader(
+      "Cache-Control",
+      "public, s-maxage=43200, stale-while-revalidate=43200"
+    );
+
+    const blockRate = await getBlockRate();
+    res.status(200).json({ blockRate30Day: blockRate.toFixed(4) });
+  } catch (err) {
+    res.status(404).json({ error: "failed to fetch data" });
```

Comment:
I believe the convention is:
status 500 - internal server error
json({ message })

if typescript doesnt like that, you can try typing the res on line 11 like this
```ts
interface ResponseData {
  blockRate30Day: string;
};
interface ErrorObj {
}


res: NextApiResponse<ResponseData | ErrorObject>
```

Reply:
this is also more handy because then we can export and use the `ResponseData` type in our apps when we fetch from this endpoint, and it wont complain about being an optional property

Reply:
awesome TypeScript is so cool!